### PR TITLE
Add readme

### PR DESCRIPTION
**Pull request by sdonkov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+![GitHub last commit](https://img.shields.io/github/last-commit/sdonkov/dawg)
+
+# DAWG
+
+Try to implement DAWG (Directed Acyclic Word Graph) also known as DAFSA is a library for computing Deterministic Acyclic Finite State Automata.
```
Comment (dmitankin) [inline README.md]: This is a really weird sentence, perhaps removing "Try to implement" will improve it...

File: README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+DAFSA are data structures derived from tries that allow representing a set of sequences that takes the form of a directed acyclic graph with a single source vertex
+(a vertex with no incoming edges), each edge of the graph is labeled by a letter or symbol, and each vertex has at most one outgoing edge for each possible letter or symbol.
+The strings represented by the DAFSA are formed by the symbols on paths in the graph from the source vertex to any sink vertex (a vertex with no outgoing edges).
+A deterministic finite state automaton is acyclic if and only if it recognizes a finite set of strings.
```
Comment (dmitankin) [inline README.md]: I'd suggest avoid putting anything here that you are not able to understand well enough to explain it to someone.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+## GOAL
+I'm researching data structures to improve my programming skills
+
+## Installation
+
+Install Maven, running unit and performance tests from the command line.
```
Comment (dmitankin) [inline README.md]: Not sure what this means. The point is to provide instructions on how to build the code and eventually contribute to it, i.e. these are instructions for developers. Place the commands that will trigger unit tests and performance tests here.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+## GOAL
+I'm researching data structures to improve my programming skills
+
+## Installation
+
+- Install Maven
+  - [Install Maven on Windows](https://www.baeldung.com/install-maven-on-windows-linux-mac#installing-maven-on-windows)
+  - [Install Maven on Linux](https://www.baeldung.com/install-maven-on-windows-linux-mac#installing-maven-on-linux)
+  - [Install Maven on Mac OSX](https://www.baeldung.com/install-maven-on-windows-linux-mac#installing-maven-on-mac-os-x)
```
Comment (dmitankin) [inline README.md]: No need for links, I guess. If you really insist on having one, use the official - https://maven.apache.org/install.html

File: README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+![💻DAWG](https://user-images.githubusercontent.com/102171106/214369779-21e03682-825a-441a-96ea-a7a4af98b61b.png)
```
Comment (dmitankin) [inline README.md:1]: Where did this picture come from?

Comment (sdonkov) [reply README.md:1]: https://banner.godori.dev/

**Review by dmitankin — APPROVED** [review]

**Pull request merged**